### PR TITLE
Mollie payment system: subscriptions, dunning, deferred Pro signup (#2, #70)

### LINKED ISSUE #2: [P0][feature] Implement Mollie billing integration (subscriptions + upgrade flow)
## Problem
Schema tables `subscriptions`, `invoices`, `platform_users.mollie_customer_id` exist (migration 000031+) and `internal/plans/enforcement.go` enforces plan limits — but there is no billing integration. Consequences:
- No way to collect payment from paying users
- No endpoint to upgrade a project free → pro
- No Mollie webhook handlers for subscription lifecycle (`subscription.created`, `subscription.canceled`, `payment.failed`)
- Console `/account` page has zero billing/payment UI
- Plan enforcement rejects free-plan users hitting pro limits, but there's no path out

MVP plan §6.4 requires Mollie integration.

## Acceptance
- [ ] `internal/billing/mollie.go` — client, customer create, subscription create/cancel
- [ ] `POST /platform/projects/{id}/billing/upgrade` — creates Mollie checkout, returns redirect URL
- [ ] `POST /platform/webhooks/mollie` — verifies signature, updates `subscriptions` + `projects.plan`
- [ ] Dunning: on `payment.failed` → email user, grace period, downgrade after N days
- [ ] Console: new `/p/{id}/billing` route with plan card, invoice list, card update
- [ ] Uses Mollie (EU, Netherlands) — sovereignty compliant

## Sovereignty
Mollie is headquartered in Amsterdam, NL — compliant with CLAUDE.md rule (no US cloud services). Confirm data residency config.

**Priority**: P0 — can't monetize without it.

### PR DESCRIPTION
**Pull request by STGime**
Closes #2, #70, #157, #158, #159, #160, #161, #162.

Schema tables existed since migration 000001 (`subscriptions`, `invoices`, `platform_users.mollie_customer_id`). This PR ships the code that turns them into a working payment system end-to-end.

### Backend
- **`internal/billing/mollie.go`** — typed Mollie REST v2 client, zero new deps. 5 calls: `EnsureCustomer`, `CreateFirstPayment` (captures mandate via `sequenceType=first`), `CreateSubscription`, `CancelSubscription`, `GetPayment`. Plus `VerifyWebhookSignature` (HMAC-SHA256 against `MOLLIE_WEBHOOK_SECRET`).
- **`internal/billing/service.go`** — orchestrates DB + Mollie. `StartUpgrade`, `StartUpgradeForPending` (#70 path), `ApplyPaymentEvent` (webhook reconciliation), `Cancel`. Audit log row on every state transition.
- **`internal/billing/dunning.go`** — 3-day grace state machine + hourly sweeper. `failed payment → grace → downgrade` and `cancel → cancel_at_period_end → period elapsed → downgrade`. Idempotent.
- **`internal/billing/handler.go`** — 5 HTTP routes. Webhook always returns 200 so Mollie doesn't loop on retries; verifies via signature + Mollie API roundtrip before applying any state.
- **Migration 000055** — `cancel_at_period_end` + `grace_until` on `subscriptions` + two partial indexes for the sweep.
- **Migration 000056** — new `pending_projects` table (24h TTL, unique-per-owner-slug) for the #70 fix.
- **5 new audit action constants**: `billing.subscription.created/.cancelled`, `payment.succeeded/.failed`, `plan.changed`.

### #70 fix — defer Pro project creation until paid
`HandleCreateProject` now takes a `CheckoutStarter`. When `plan='pro'` AND Mollie is configured, the project is **NOT created immediately** — instead `pending_projects` reserves the slug, Mollie checkout starts, and the response is `{ status: 'pending_payment', pending_project_id, checkout_url }`. The real `public.projects` row is materialised by the webhook handler on `payment.paid` via a `ProjectProvisioner` callback (avoids a `tenant → billing` import cycle). Free signup flow is unchanged.

### Console
- **New `/p/[id]/billing` page** — subscription card, status badge, invoice list, Upgrade button, Cancel modal with grace-period explanation, `?autostart=1` deep-link support for the pricing-page CTA.
- **4 new API methods** + `SubscriptionView` / `Invoice` / `PendingProjectResponse` types. `createProject` return type widened to a union.

### Drive-by
- `sdk/js/README.md` Testing section documenting the thenable `QueryBuilder` pattern + `MockQueryBuilder` example (same bug another consumer just hit in their tests).

## Routes
| Method | Path | Auth |
|---|---|---|
| `POST` | `/platform/projects/{id}/billing/checkout` | admin |
| `GET`  | `/platform/projects/{id}/billing/subscription` | viewer |
| `POST` | `/platform/projects/{id}/billing/cancel` | admin |
| `GET`  | `/platform/projects/{id}/billing/invoices` | viewer |
| `POST` | `/webhooks/mollie` | public (signature-verified) |

## Tests
- **9 unit tests** against an httptest-mocked Mollie (`mollie_test.go`) — signature verification, error propagation, all client methods, amount parsing.
- **5 DB-backed tests** (`-short` skips) — free→pending→active, failed→grace, sweeper downgrade after grace expiry, idempotency on duplicate webhooks, fresh project returns synthetic free view.
- `internal/gateway/integration_test.go` updated to thread `nil` `CheckoutStarter` (pre-Mollie path preserved).

## Verification
- `go test -short ./...` → all 26 packages green incl. new `internal/billing`
- Console `npm run build` → 0 errors

## Operator action required before this is "live"

1. Sign up for a Mollie account, generate `test_*` API key
2. Generate webhook signing secret (`openssl rand -hex 32`)
3. Populate `eurobase-secrets` k8s Secret with `MOLLIE_API_KEY` + `MOLLIE_WEBHOOK_SECRET`
4. Configure webhook URL in Mollie dashboard: `https://api.eurobase.app/webhooks/mollie`
5. Smoke test with Mollie's test cards
6. Cutover from `test_` to `live_` key when ready to take real money

Full procedure in `docs/runbooks/billing-mollie.md`.

## Test plan

- [x] `go test -short ./...` — all packages green
- [x] Console builds
- [ ] CI green on this PR
- [ ] After merge: populate vault keys, configure Mollie webhook, smoke test against a test project (the runbook lists exact steps)
- [ ] Verify `/platform/projects/{id}/compliance/dpa-report` lists Mollie when a subscription row exists

## Out of scope (deferred follow-ups)

- Cross-page "Pro pending" banner on every project header (billing page surfaces state)
- CLI `eurobase admin billing reconcile <project>` (curl path documented)
- Mollie hosted customer-portal embed (we redirect to fresh checkout for card updates)
- Annual billing tier
- Edge function metering (#128), per-org Team tier (#131)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/billing/service.go [hunk 1]
```diff
@@ -0,0 +1,494 @@
+package billing
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
```
Comment (STGime) [inline internal/billing/service.go:6]: **Issue (medium):** `ProPriceEUR = "9.00"` here but the pricing page (PR #133) and the onboarding card both say **€19/mo**. One of these is wrong. If Pro is €19, this constant and `ProAmountCents = 900` need to be `"19.00"` and `1900`. If Pro is €9, the pricing page is wrong. Either way, shipping mismatched amounts means the first real customer sees a different price at checkout than what they were shown — that's a billing compliance issue.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by STGime — COMMENTED** [review]
Comprehensive billing integration — well-architected with proper security layering and a thorough runbook. I read the Mollie client, service layer, dunning sweeper, webhook handler, handler routes, console billing page, tests, and runbook. Key findings:

### Issues to fix before merge

1. **Price mismatch (medium):** `ProPriceEUR = "9.00"` / `ProAmountCents = 900` in `service.go` but the pricing page (#133) and onboarding card both show **€19/mo**. The first real customer will see a different amount at Mollie checkout than what the marketing page promised. Fix to `"19.00"` / `1900` if Pro is €19, or update the pricing page if Pro is €9. Inline comment on this.

2. **Webhook body double-read (bug):** In `HandleWebhook`, `io.ReadAll(r.Body)` consumes the body for signature verification (line ~968). Then `r.ParseForm()` (line ~988) tries to re-read `r.Body` — but it's already drained, so `PostFormValue("id")` always returns `""`. The JSON fallback path below catches it (parses `raw`), but the form-parsing path is dead code. Fix: replace with `url.ParseQuery(string(raw))` to extract the payment ID from the already-read bytes.

### Architecture — well done

- **Webhook security:** Triple-verification (HMAC signature → Mollie API roundtrip → idempotent state machine) means even a replayed webhook can't corrupt state. Constant-time compare on the signature.  Always returning 200 prevents retry storms.
- **Dunning state machine:** `failed → grace (72h) → downgrade` and `cancel → pro_until_period_end → period elapsed → downgrade`. Idempotent WHERE clauses, hourly sweep. Clean separation from the webhook handler.
- **#70 fix:** Deferred project creation via `pending_projects` table + `ProjectProvisioner` callback avoids the tenant→billing import cycle. Failed/abandoned checkouts delete the reservation so the slug is freed.
- **`EnsureCustomer` idempotency:** Two concurrent calls for the same user both end up with the same `mollie_customer_id` — the second caller sees the cached value from the `WHERE ... IS NULL` guard.
- **Runbook quality:** First-time setup, smoke test, manual reconciliation, refund handling, common failure modes. Production-ready.

### Tests

- 9 unit tests against mocked Mollie — signature verification, all client methods, error surfacing, amount parsing.
- 5 DB-backed tests — state machine transitions, sweeper, idempotency on duplicate webhooks.
- `euroStringToCents` edge cases including truncation at 2 decimals. Good.

### Console

- Billing page with subscription card, status badges, invoice table, Upgrade/Cancel buttons, cancel modal with grace-period explanation, `?autostart=1` deep-link.
- Clean status labels and badge colors for all 6 states.

Fix the price mismatch and the webhook body double-read before merge. Everything else is solid.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/billing/service.go (hunk 1 above)
Comment (STGime) [reply internal/billing/service.go:6]: Fixed in dc45e93 — `ProPriceEUR` is now `"19.00"` and `ProAmountCents` is `1900` to match the pricing page, onboarding card, and projects list. Confirmed by grep: the marketing site and console all reference €19/mo consistently. Added a comment block above the constants explaining the single-source-of-truth contract until we move pricing into `plan_limits` for a Team tier.

**Comment by STGime** [comment]
Both review issues fixed in dc45e93.

**1. Price mismatch (€9 → €19)** — `ProPriceEUR` and `ProAmountCents` updated; billing page subhead text updated. Confirmed by grep that the pricing page, onboarding card, and projects list all reference €19/mo consistently.

**2. Webhook form-parse bug** — Real bug; would have made every production Mollie webhook a no-op since their default Content-Type is form-urlencoded. Fix: replaced `r.ParseForm()` (which reads the already-drained `r.Body`) with `url.ParseQuery(string(raw))` on the bytes we already have.

Added 3 regression tests in `internal/billing/handler_test.go`:
- `TestHandleWebhook_ExtractsPaymentIDFromFormBody` — drives a signed form-urlencoded request through HandleWebhook, asserts Mollie's GetPayment endpoint is hit (which only happens if the payment ID was successfully extracted). This would have caught the original bug.
- `TestHandleWebhook_RejectsMissingSignature` — unsigned → no Mollie API roundtrip
- `TestHandleWebhook_RejectsMisSignedBody` — mis-signed → no Mollie API roundtrip

All 26 Go packages still green under `go test -short`. Ready for re-review.

**Review by STGime — COMMENTED** [review]
Both issues fixed cleanly in dc45e93:

1. **Price** — `ProPriceEUR = "19.00"`, `ProAmountCents = 1900`. Matches pricing page.
2. **Webhook form-parse** — `url.ParseQuery(string(raw))` replaces the broken `r.ParseForm()` on the drained body. 3 new regression tests cover the fix.

LGTM — ready to merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by STGime** [comment]
Closing as stale. This PR was opened 2026-05-25 and never merged. The Mollie billing system it proposed was ultimately delivered through a series of smaller, focused PRs rather than the single 2-commit landing here:

- #443 Buyer profile capture before Mollie checkout (Estonian VAT §37)
- #465 Cutover — \`MOLLIE_ENV\` flipped to \`live\` in prod (2026-08-25)
- #466 Post-checkout redirect 404 fix (\`/projects/{id}/billing → /p/{id}/billing\`)
- #467 Cancel/expire on first payment frees local sub row for retry
- #496 Table-driven test for \`markPaymentFailure\` (closed #468)
- #497 Bump Pro to €25 + explicit Free-tier non-commercial framing
- ...plus the initial code drop that predates this PR

Current prod state (verified in \`deploy/k8s/gateway.yaml\`):
- \`BILLING_ENABLED=true\`
- \`MOLLIE_ENV=live\`
- \`internal/billing/mollie/\` package shipped (client.go, errors.go, types.go, tests)
- \`public.billing_profiles\` shipped for B2B invoice fields (see #414 close)
- Dunning + downgrade sweeper live
- Console \`/p/[id]/billing\` page + \`/p/[id]/billing/checkout\` handler live
- Webhook \`POST /webhooks/mollie\` signature-verified and receiving

No content salvage from this branch — the code that shipped is materially different from the diff here.
